src/assets/Generator.Shared/GeneratorPageableHelpers.cs

```diff
@@ -41,7 +41,20 @@ public static AsyncPageable<T> CreateAsyncPageable<T>(
         {
             return new PageableHelpers.AsyncPageableWrapper<T>(new PageableHelpers.PageableImplementation<T>(null, createFirstPageRequest, createNextPageRequest, valueFactory, pipeline, clientDiagnostics, scopeName, itemPropertyName, nextLinkPropertyName, null, cancellationToken, null));
         }
-
+        public static AsyncPageable<T> CreateAsyncPageable<T>(
+            Func<int?, HttpMessage>? createFirstPageRequest,
+            Func<int?, string, HttpMessage>? createNextPageRequest,
+            Func<JsonElement, T> valueFactory,
+            ClientDiagnostics clientDiagnostics,
+            HttpPipeline pipeline,
+            string scopeName,
+            string? itemPropertyName,
+            string? nextLinkPropertyName,
+            object defaultPageSize,
```

Comment:
This should be `int`, we don't allow run-time casting during generation.

Reply:
Fixed